rasa/shared/core/domain.py

```diff
@@ -721,11 +738,9 @@ def __init__(
             session_config: Configuration for conversation sessions. Conversations are
                 restarted at the end of a session.
         """
-        self.entities, self.roles, self.groups = self.collect_entity_properties(
-            entities
-        )
+        self.entity_properties = self.collect_entity_properties(entities)
```

Comment:
Optional (mainly thinking out loud): If this attribute doesn't require access outside `Domain`, should we make it private?

Reply:
I'm not that familiar with the benefits of private methods in python and I'm only seeing very limited (mostly representation related) private methods in the code base.  Making this one private would imply many others should be too, which might be, but probably out of scope for this pr.

Reply:
I was actually suggesting to transform this into a private attribute preceded by a single underscore, which only signals to another developer that this shouldn't be accessed or set outside the Domain class.  There are many examples through the codebase of such attributes. If this was preceded by double underscore, then it would be strongly protected from being modified.

Reply:
Ah! Sorry I misread and went looking for private methods instead of attributes. That does make sense.